libswift/Sources/SIL/Builder.swift

```diff
@@ -0,0 +1,61 @@
+//===--- Builder.swift -  Building and modifying SIL ----------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SILBridging
+
+/// A utility to create new instructions at a given insertion point.
+public struct Builder {
+  let insertionPoint: Instruction
+  private let passContext: BridgedPassContext
+
+  private var bridgedInsPoint: BridgedInstruction { insertionPoint.bridged }
+  
+  private func notifyInstructionsChanged() {
```

Comment:
To minimize the usage of the bridging API. `notifyInstructionsChanged()` will be called from a lot of places in the builder and here it's a central place where `PassContext_notifyChanges` is called. It makes it easier in case we change the bridging layer at some time.

Reply:
Right, the C API needs to be called in a central place. That's why I expect that the Swift code to need a PassContext struct to encapsulate the C API. There are more ways to change instructions than just creating new ones